src/tcutility/molecule.py

```diff
@@ -176,66 +149,149 @@ def guess_fragments(mol: plams.Molecule) -> Dict[str, plams.Molecule]:
-                elif isinstance(indx, str) and '-' in indx:
-                    indices.extend(range(int(indx.split('-')[0]), int(indx.split('-')[1]) + 1))
+                elif isinstance(indx, str) and "-" in indx:
+                    indices.extend(range(int(indx.split("-")[0]), int(indx.split("-")[1]) + 1))
                 else:
-                    raise ValueError(f'Fragment index {indx} could not be parsed.')
+                    raise ValueError(f"Fragment index {indx} could not be parsed.")
 
             [fragment_mols[frag_name].add_atom(mol[i]) for i in indices]
-            fragment_mols[frag_name].flags = {'tags': set()}
-            if f'charge_{frag_name}' in mol.flags:
-                fragment_mols[frag_name].flags['charge'] = mol.flags[f'charge_{frag_name}']
-            if f'spinpol_{frag_name}' in mol.flags:
-                fragment_mols[frag_name].flags['spinpol'] = mol.flags[f'spinpol_{frag_name}']
+            fragment_mols[frag_name].flags = {"tags": set()}
+            if f"charge_{frag_name}" in mol.flags:
+                fragment_mols[frag_name].flags["charge"] = mol.flags[f"charge_{frag_name}"]
+            if f"spinpol_{frag_name}" in mol.flags:
+                fragment_mols[frag_name].flags["spinpol"] = mol.flags[f"spinpol_{frag_name}"]
 
         return result.Result(fragment_mols)
 
     # second method, check if the atoms have a frag= flag defined
-    fragment_names = set(atom.flags.get('frag') for atom in mol)
+    fragment_names = set(atom.flags.get("frag") for atom in mol)
     if len(fragment_names) > 0:
         fragment_mols = {name: plams.Molecule() for name in fragment_names}
         for atom in mol:
             # get the fragment the atom belongs to and add it to the list
-            fragment_mols[atom.flags.get('frag')].add_atom(atom)
+            fragment_mols[atom.flags.get("frag")].add_atom(atom)
 
         for frag in fragment_names:
-            fragment_mols[frag].flags = {'tags': set()}
-            if f'charge_{frag}' in mol.flags:
-                fragment_mols[frag].flags['charge'] = mol.flags[f'charge_{frag}']
-            if f'spinpol_{frag}' in mol.flags:
-                fragment_mols[frag].flags['spinpol'] = mol.flags[f'spinpol_{frag}']
+            fragment_mols[frag].flags = {"tags": set()}
+            if f"charge_{frag}" in mol.flags:
+                fragment_mols[frag].flags["charge"] = mol.flags[f"charge_{frag}"]
+            if f"spinpol_{frag}" in mol.flags:
+                fragment_mols[frag].flags["spinpol"] = mol.flags[f"spinpol_{frag}"]
 
         return result.Result(fragment_mols)
 
 
+# =============================================================================
+# Writing Functions for Molecule objects ======================================
+# =============================================================================
+
+
+def _xyz_format(mol: plams.Molecule) -> str:
+    """Returns a string representation of a molecule in the xyz format, e.g.:
```

Comment:
Is this docstring correct? Looks the same as for the amv format

Reply:
Ups, will change it